content/pytorch/concepts/nn/activation-functions/activation-functions.md

```diff
@@ -0,0 +1,199 @@
+---
+Title: 'PyTorch nn Activation Functions'
+Description: 'Set of pre-built activation functions that are essential for building neural networks.'
+Subjects:
+  - 'AI'
+  - 'Machine Learning'
+  - 'Computer Science'
+Tags:
+  - 'AI'
+  - 'Neural Networks'
+CatalogContent:
+  - 'intro-to-py-torch-and-neural-networks'
+  - 'py-torch-for-classification'
+---
+
+The **`nn Activation Functions`** in Pytorch are a set of pre-built activation functions that are essential for building neural networks. Activation functions are mathematical equations that determine the output of a neural network. The function is attached to each neuron in the network and determines whether it should be activated or not, based on whether each neuron's input is relevant for the model's prediction. Activation functions also help normalize the output of each neuron to a range between 1 and 0 or between -1 and 1.
+
+Activation functions are used to introduce non-linearity to the neural network. Common activation functions include `ReLU`, `ReLU6`, `Leaky ReLU`, `Sigmoid`, `Tanh`, and `Softmax`. Activation functions are applied to the output of each neuron in the network.
+
+## ReLU (Rectified Linear Unit)
+
+ReLU is a popular activation function that returns the input if it is positive, and zero otherwise. It is defined as:
+
+```pseudo
+f(x) = max(0, x)
+```
+
+ReLU is widely used in deep learning models due to its simplicity and efficiency. It helps the model learn complex patterns in the data by introducing non-linearity.
+
+```py
+import torch.nn as nn
+
+# Define a neural network model with ReLU activation function
+class SimpleNN(nn.Module):
+  def __init__(self):
+    super(SimpleNN, self).__init__()
+    self.fc1 = nn.Linear(10, 5)
+    self.fc2 = nn.Linear(5, 1)
+    self.relu = nn.ReLU()
+
+  def forward(self, x):
+    x = self.relu(self.fc1(x))
+    x = self.fc2(x)
+    return x
+
+# Create a model instance
+model = SimpleNN()
+```
+
+## ReLU6
+
+ReLU6 is a variation of the ReLU activation function that returns the input if it is positive and less than 6, and zero otherwise. It is defined as:
+
+```pseudo
+f(x) = min(max(0, x), 6)
+```
+
+ReLU6 is useful when you want to limit the output of the activation function to a specific range, such as between 0 and 6.
+
+```py
+import torch.nn as nn
+
+# Define a neural network model with ReLU6 activation function
+class SimpleNN(nn.Module):
+  def __init__(self):
+    super(SimpleNN, self).__init__()
+    self.fc1 = nn.Linear(10, 5)
+    self.fc2 = nn.Linear(5, 1)
+    self.relu6 = nn.ReLU6()
+
+  def forward(self, x):
+    x = self.relu6(self.fc1(x))
+    x = self.fc2(x)
+    return x
+
+# Create a model instance
+model = SimpleNN()
+```
+
+## Leaky ReLU
+
+Leaky ReLU is a variation of the ReLU activation function that returns the input if it is positive, and a small fraction of the input otherwise. It is defined as:
```

Comment:
```suggestion
Leaky ReLU is a variation of the ReLU activation function that returns the input if it is positive, and a small fraction of the input otherwise. It is mathematically defined as:
```